daemon/team.go

```diff
@@ -64,4 +98,40 @@ func (d *daemon) RestartTeamLab(req *pb.RestartTeamLabRequest, resp pb.Daemon_Re
 	}
 
 	return nil
-}
\ No newline at end of file
+}
+
+// suspendTeams function will check all teams in all events
+// then suspend their resources if lastAccessTime is higher than INACTIVITY_DURATION
+func (d *daemon) suspendTeams() error {
+	var suspendError error
+	log.Info().Msg("Suspend teams called ! ")
+	events := d.eventPool.GetAllEvents()
+	now := time.Now()
+
+	for _, e := range events {
+		for _, t := range e.GetTeams() {
+			if !t.LastAccessTime().IsZero() {
+				difference := math.Round(now.Sub(t.LastAccessTime()).Minutes()) // get in rounded format in hours
+				if difference > INACTIVITY_DURATION {
+					lab, ok := e.GetLabByTeam(t.ID())
+					if !ok {
+						return UnknownTeamErr
+					}
+					// check if it is already suspended or not.
+					lab.InstanceInfo()
+					for _, instanceInfo := range lab.InstanceInfo() {
```

Comment:
why calling `lab.InstanceInfo()` twice ?

Reply:
hmm, I really did not notice it, probably it is something forgotten from development process.